src/App.tsx

```diff
@@ -15,48 +19,123 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = 'none',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+): string[] {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort(
+        (prevGood, currGood) => prevGood.localeCompare(currGood),
+      );
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort(
+        (prevGood, currGood) => prevGood.length - currGood.length,
+      );
+      break;
+
+    default:
+      break;
+  }
+
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const reverse = () => {
+    setIsReversed(current => !current);
+  };
+
+  const sortByType = (type: SortType) => {
+    setSortType(type);
+  };
+
+  const reset = () => {
+    setSortType(SortType.NONE);
+    setIsReversed(false);
+  };
+
+  const showResetButton = sortType !== SortType.NONE || isReversed;
```

Comment:
Naming conventions for boolean variables and functions are important because they help make code more readable and easier to understand.

Boolean variables, which represent true or false values, are often used to store the state of a condition in a program. Starting their names with "is", "should", or "does" helps to indicate that they are boolean variables and makes it clear that their value represents a true or false condition.

For example, if you have a boolean variable that represents whether a user is logged in or not, you might name it "isLoggedIn" or "shouldLogin". This makes it clear that the variable represents a true or false condition related to the login state of the user.

Functions, on the other hand, are blocks of code that perform a specific action or task. Starting their names with a verb helps to indicate that they are functions and makes it clear that they are performing an action.

For example, if you have a function that adds two numbers together, you might name it "addNumbers". This makes it clear that the function performs an action and helps to distinguish it from other types of variables or objects in the code.

In general, following these naming conventions can help make your code more readable and easier to understand for other developers who may work on the same codebase in the future.